provider/core/src/request.rs

```diff
@@ -129,7 +129,8 @@ impl Ord for DataIdentifierCow<'_> {
         self.marker_attributes
             .as_str()
             .cmp(other.marker_attributes.as_str())
-            .then_with(|| self.locale.total_cmp(&other.locale))
+            .then_with(|| self.locale.langid.total_cmp(&other.locale.langid))
+            .then_with(|| self.locale.keywords.cmp(&other.locale.keywords))
```

Comment:
Issue: `self.locale.langid` and `self.locale.keywords` are private fields of DataLocale and they should not be being read by DataIdentifier; when we change DataLocale to have a different set of private fields, it should stay that way. I'm not sure why you removed `DataLocale::total_cmp`.

Reply:
`DataIdentifier` and `DataLocale` are very tightly coupled. I'm removing `DataLocale::total_cmp` because nobody should be putting a `DataLocale` in a `BTreeMap`, downstream code should always deal with `DataIdentifier`s.

Reply:
If the fields are private, it's not correct for code from a different struct to read them. You could make `DataLocale::total_cmp` be `pub(crate)` if you don't want people using it. But I won't block the PR on this.

Reply:
The fields are module-private, and are used from code in the same module.